core/chains/evm/txm/types/transaction.go

```diff
@@ -0,0 +1,194 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/google/uuid"
+	"gopkg.in/guregu/null.v4"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+	clnull "github.com/smartcontractkit/chainlink-common/pkg/utils/null"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/gas"
+)
+
+type TxState string
+
+const (
+	TxUnstarted   = TxState("unstarted")
+	TxUnconfirmed = TxState("unconfirmed")
+	TxConfirmed   = TxState("confirmed")
+
+	TxFatalError = TxState("fatal")
+	TxFinalized  = TxState("finalized")
+)
+
+type Transaction struct {
+	ID                uint64
+	IdempotencyKey    *string
+	ChainID           *big.Int
```

Comment:
Is the chainID stored per tx for easier logging?

Reply:
ChainID might be more of a QoL improvement here, but it's necessary once we have a DB implementation.